### DISCUSSION TITLE
GitHub Sponsors will stop accepting PayPal on February 23, 2023

### DISCUSSION THREAD

**Discussion by lupilabs** [community/community#45183] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

WTF is wrong with U.S. companies?!
A credit card is the most insecure and privacy-poor payment method.
If Github will do so, I will not be able to sponsor anymore.

There are dozens of other payment methods - there MUST be an alternative!

**Comment by Revadike** [comment]
![NoOneLikesItSimonGIF](https://user-images.githubusercontent.com/4411977/214214501-eefa61e7-aab6-46b1-a27b-6bb7d45547f2.gif)

**Reply by iamnormatov** [reply]
It's funny

**Comment by GrizzlyAK** [comment]
I will no longer be sponsoring here on Github without Paypal either, unfortunately. I can't imagine what idiot thought this was a good idea. Probably because PayPal took a small percentage of the fee for processing payments. Yeah, well, just wait until those Visa/MC processing fees start rolling in, especially when people use their "miles" cards that fetch premium fees from merchants (and ultimately, consumers).

**Reply by jlsjonas** [reply]
Paypal is sadly enough notorious for an extremely high processing fee (often more than 5%), but this is indeed not the solution... :/
Informing, providing alternatives & discouraging the use of less favorable providers would be such a better way to deal with the situation... Should we be surprised though?

**Comment by dennis1248** [comment]
Yeah, shame this is getting dropped. For us European users, where it is common to have a debit card but not a credit card, PayPal is often the only available payment option. This change makes it so that I will be unable to continue sponsoring people via GitHub.

And I totally agree with credit cards being a frankly stupidly insecure method of payment.

**Reply by dennis1248** [reply]
The minimal effort fix for this issue for me personally would be for GitHub to just send me an invoice once per month and allow me to deposit the amount due to an IBAN number.

Although it would be highly preferred if they were to support both local payment methods and Euro direct debit.

**Reply by Mitsunee** [reply]
seconding this, it's super uncommon to have a credit card here as they both require a significant amount of income (that I personally don't even have) AND have a monthly fee. I'm probably going to lose both of my sponsors and will have to go back to paying for my projects out of my savings until I can find a new source of income.

**Reply by dennis1248** [reply]
@Mitsunee There is Liberapay which is EU-based. But that does mean no fancy badge or any other visual indication of being a sponsor. The visibility which GitHub provides is a large plus.

**Reply by dtugend** [reply]
@Mitsunee Also checkout https://opencollective.com/ we had to decide between LiberaPay and them and in the end chose them, but check what suits you better of course.

**Reply by Jackenmen** [reply]
You **CAN** use a debit card (or really any type of bank card) to pay as well... GitHub should probably rephrase it in their UI to make it clear but pretty much everywhere you're asked for credit card details, you can use a debit card, and usually other types of bank cards as well.

I'm not sure why everyone is saying that payment with bank cards is very insecure... It's generally very secure due to the ability to make a dispute with your card provider using the Chargeback procedure. For people who aren't convinced by that, there are even virtual card services that they can use instead.

Meanwhile, PayPal's service fees are very high in comparison to competitors such as Stripe, their terms for companies aren't ideal, and their terms for consumers aren't that much better either.

**Reply by bodograumann** [reply]
I checked your statement, @Jackenmen, and it is **not true**.
I cannot use my maestro card for paying github sponsors.
The form just says "invalid card number".

**Reply by dennis1248** [reply]
@Jackenmen Debit cards which support credit card payment methods, yes. Yet these are relatively rare. They are currently slowly starting to roll out these new payment methods called Debit Mastercard and Visa Debit across Europe, but it will likely take up to a decade for most people to have these new cards.

**Reply by virtualritz** [reply]
> @Jackenmen Debit cards which support credit card payment methods, yes. Yet these are relatively rare. They are currently slowly starting to roll out these new payment methods called Debit Mastercard and Visa Debit across Europe, but it will likely take up to a decade for most people to have these new cards.

That's simply not true. I had such cards with all my bank accounts since almost a decade in Germany. Fidor, N26, Revolut.

**Reply by enc** [reply]
In germany there is a special debit card. It is not a credit card.

These days a lot of credit cards wort based on debit and not credit, but they are still "credit cards". The german system is knowing a giro card. These have much lower transaction costs usually.

**Reply by hmans** [reply]
Let's not forget that the real issue here is the extremely poor communication by GitHub. If you have a significant number of sponsors that are somewhat evenly distributed over the planet, count on up to 50% of them going away this month. GitHub should at least have given project owners some time to set up alternatives and get in touch with their sponsors first. I can't take any of the talk from GitHub about wanting to help people build sustainable income from their open source work seriously anymore.

**Reply by GrizzlyAK** [reply]
The problem with debit cards, at least in the US, is if the card number is stolen and someone uses it, your money is immediately gone from your bank account (unlike a credit card), and it is much harder and takes longer to get it back. In the meantime, your cash is "gone". That could be everything you had in the account. 

I rarely use mine anymore.

**Reply by mantono** [reply]
> I checked your statement, @Jackenmen, and it is **not true**. I cannot use my maestro card for paying github sponsors. The form just says "invalid card number".

Maestro does not support recurring payments, or payments done as "merchant initiated transactions". This is a limitation with them and not GitHub. Most VISA and MasterCard issued cards should work fine tough. I have paid in the past (monthly recurring fee) for GitHub with a debit card, so you should not need a credit card. It is just common to say "credit card" when you mean either credit card or debit card.

That being said, I still think that PayPal is the superior alternative from a customer's point of view, and it is sad that it is being dropped.

EDIT: **I just tried changing from PayPal, back to my card that I have used in the past, and it was not accepted.**

**Reply by redjr16** [reply]
Maybe so, but debit is different.  it's simply not wise to use your debit card for online transactions!  If that debit card is ever hacked, goodbye to your checking account balance.  I've made it a practice (for years now), to never give up my debit card # on the internet.  Mine isn't anywhere near my computer or phone!  Period.  I learned this about about 3 decades ago, when all this online purchasing was just getting started and bank affiliated debit cards were new.  The safest way to pay is to use a refillable Visa or Mastercard **exclusively** for online purchases.   Carry a low limit ($1,000) and refill when necessary.

**Comment by unitycoder** [comment]
Then forced to stop sponsoring also.

> "We're focused on providing the best Sponsors experience for everyone involved."

Narrator: But it didn't work.

Sounds more like github/microsoft later introducing their own paypal competitor?

**Comment by andrewreeman** [comment]
Just cancelled all my sponsorships because of this, sadly. Really don't want to store payment details on another site.

**Comment by Guacam-Ole** [comment]
I wonder if they at least are honest enough to provide monthly statistics so we can at least say "told you so" after the numbers drop next month

**Reply by hmans** [reply]
I'm already seeing incoming sponsorship cancellations. The damage is already done.

**Comment by ghost** [comment]
PayPal is an absolutely horrible company, but it's unfortunately a monopoly that so many people have no choice but to use.

I don't know what caused this change, but I wouldn't be surprised if PayPal themselves did something to bring about this change one way or another.

Either way, I hope the fallout isn't too bad for the people relying on sponsorships.

For people concerned over the security of using credit card payments online, I strongly recommend looking into a service such as Privacy that allows you to generate an unlimited number of virtual cards, with controllable spending limits and the ability to pause and close them at will.

I believe they only operate in the U.S. at the moment sadly, but if you can use it, it's a great service.

Your card company may also provide an option to generate virtual cards, though it may be annoying to actively utilize.

**Reply by hmans** [reply]
> I don't know what caused this change, but I wouldn't be surprised if PayPal themselves did something to bring about this change one way or another.

Yes, I've seen some people mention that GitHub was, strictly speaking, violating PayPal's ToS by using it in this program, and that PayPal now pulled the plug. (I can't confirm, but whatevs.) Nobody loves PayPal, so good riddance, _but_ the really bad thing here is how this was communicated. There are people who _depend_ on their GH Sponsors income, and getting a short non-committal email telling them essentially that half of it is going to go away next month is super super bad. In any other sort of business relationship there would have been a 6-month notice and some type of migration support. The message here is that GitHub really doesn't give much of a fuck about people using this program.

**Reply by Guacam-Ole** [reply]
I think in some regions outside the US it is more like 80% loss. In germany for example Credit Card usage is still quite rare, especially online

**Comment by philipisik** [comment]
This is very bad news for small teams like us who desperately need sponsorship to continue working on their open source project. 😢

**Reply by redjr16** [reply]
Yes.  This policy will devastate individuals also who are trying to make a go of it and their sponsors simply quit supporting them.  I have a decision to make....

**Comment by Phoenix616** [comment]
Please provide ways to sponsor without credit cards. Lots of people were I live (Germany) do not have them and PayPal allowed payment without cards too!

At least support SEPA, the standard way to pay in the EU!

**Comment by sstevo66** [comment]
How does the affect those of us that were pushed to Stripe. The announcement says Credit and Debit cards **_Only._**

**Comment by borkdude** [comment]
I turned open source into my day job in large part due to Github Sponsors.

I am trying to switch to a creditcard myself for the people that I sponsor but my card is rejected:

<img width="812" alt="Screenshot 2023-02-15 at 17 56 36" src="https://user-images.githubusercontent.com/284934/219099284-103db91d-e981-4cf9-af09-f536ebae9aaa.png">

If people have a similar experience like mine, this will cost me _a lot_ of sponsors. I'm really sad about that, but it was fun while it lasted.

**Reply by magicznyleszek** [reply]
I am coming to this thread from an official email. I was supporting a project via PayPal and now I will not be able to do that. Good job GH at "providing the best Sponsors experience for everyone involved".

**Reply by borkdude** [reply]
Lost my credit card (in my own house) once and ordered a new one. Then I re-found the old one. But since they looked almost identical, I couldn't tell the old from the new one, so I assumed it was the one with the latest expiration date and got rid of the other. 

I chose wrong and ordered a new credit card once again 🤦

**Comment by unitycoder** [comment]
support replied, that at least no need to do anything, paypal sponsoring gets automatically cancelled:

> You will not need to do anything to cancel the Sponsorships. Any Sponsorships set up with PayPal as the payment method will automatically be canceled on February 23

Any alternatives?
Patreon is not nice alternative, since they take up to 12% cut..

**Reply by razonyang** [reply]
Ko-fi may be a good alternative. https://help.ko-fi.com/hc/en-us/articles/360002506494-Does-Ko-fi-Take-a-Fee-

**Comment by lupilabs** [comment]
Thanks for the discussion and your perspectives so far. I thought I'd write again, the outcry didn't trigger any reaction or further explanation @GitHub.

I will pay for my sponsorships one year in advance via One-Time-Payment - Paypal is still activated there. 
Then you have at least a year to find new ways.

**Comment by squidfunk** [comment]
We currently have [+500 sponsors](https://github.com/sponsors/squidfunk) – I'm seriously concerned what the impact will be. Please, @github, there must be alternatives.

**Reply by borkdude** [reply]
@squidfunk Hi! Did you notice any effect yet?

https://github.com/orgs/community/discussions/45183#discussioncomment-5101093

**Reply by squidfunk** [reply]
Sadly, yes. On February 23, 4 yearly sponsorships were terminated prematurely. I have webhooks in place that add/remove collaborators on a private repo automatically, resulting in collaborators that are tied to yearly sponsorships that still run for multiple months to be removed too early. I'm already in contact with those individuals and orgs. Today, I lost 5 more monthly sponsors. If this trend continues, I expect to lose around 150 of my 500 sponsors in the next 30 days, probably around $2,000 (or even more) of monthly sponsorships.

I'm pretty sure that you can tell the difference between a user-induced and GitHub-induced sponsorship termination by checking if a __cancellation event__ was preceded by a __pending cancellation event__. If that's not the case, the sponsorship was likely terminated by GitHub. That's at least what my research suggests.

As a last measure, I contacted all sponsors yesterday via the newsletter functionality, telling them about the impending change and that I now also have a [Ko-fi](https://ko-fi.com/squidfunk) profile in place to add back PayPal support. However, I expect the churn to be significant.

This is bad.

**Comment by obahareth** [comment]
Unfortunately GitHub frequently says credit/debit cards from my country are declined, but using those same cards through PayPal always worked.

I don't know if this is affecting only sponsorships or everything, but either way people will lose the ability to pay. I hope this decision comes with a quick alternative for those people, and not just removing the ability to pay from many GitHub users.

**Reply by lupilabs** [reply]
Today they sent an email to update my "Copilot" payment method...
Byebye Copilot, Hello ChatGTP... damn, openai.com also only supports credit card payments ://

**Comment by notpushkin** [comment]
I don't like PayPal – as a non-US citizen, I've had more problems with it than with credit cards. However, I agree that we need more payment options. Maybe look into [other payment methods](https://stripe.com/us/payments/features#payment-options) accepted by Stripe?

**Comment by nikit0ns** [comment]
### **This, of course, is bad news. But there is Ko-fi may be a good alternative.** 
- https://help.ko-fi.com/hc/en-us/articles/360002506494-Does-Ko-fi-Take-a-Fee-

**Comment by GrizzlyAK** [comment]
@github-staff this really sucks, you know. Maybe it's time for the Open Source community to move their projects to GitLab.

https://www.gartner.com/reviews/market/value-stream-delivery-platforms/compare/github-vs-gitlab

**Reply by nikit0ns** [reply]
I wouldn't want to switch to GitLab. I like GitHub better

**Reply by nikit0ns** [reply]
If not PayPal, then what else?

**Comment by LE4711** [comment]
This, of course, is very very bad news. if github sticks with the decision, i will look for an alternative to evcc. Too bad!

**Comment by borkdude** [comment]
I only lost 3 sponsors so far, I expected it to be way worse, but maybe it's not over yet, or should users who paid with Paypal already be cancelled by now?

EDIT: ouch, it's not over yet, cancellations are still trickling in.

**Reply by squidfunk** [reply]
[I think we'll be losing sponsors every day now for the next 30 days](https://github.com/orgs/community/discussions/45183#discussioncomment-5101357).

__Edit__: Jup, every day. Between Feb 23 and Mar 1 we lost 30 sponsors, 10 yesterday alone.

**Reply by unitycoder** [reply]
my sponsorships are still visible, shows May 11th for the next due date.. so probably will be closed at that time.

**Reply by squidfunk** [reply]
[The damage is done](https://twitter.com/calebporzio/status/1638560981121921029). Let's hope the drain will finally stop.

**Reply by squidfunk** [reply]
It looks like it stopped, at least for me.

**Comment by Wickedlilang3l** [comment]
There are several ways to process payments online. I could act as a processor for some of y'all if necessary. I'm working on startup Global Guardian Ambassadors ... it's a company with a plan to cause huge resistance for cybercriminals. PayPal isn't as secure as other methods... there's cash app money Lion Venmo skrill Chile plaids stripe and many more.... collaboration it's the best way to solve most things. Thank everything for having me in here. For anyone assisting in developing web3 thank you.. you guys are so under appreciated. I hope I didn't break any rules. Please be patient with me every one I need a lot of help. I don't know the first thing about coding, or coding language. The passion at my core is anti human trafficking and causing a crap ton of awareness...

**Comment by LlmDl** [comment]
I lost 1/5th of my sponsors thanks to this change. Very disheartening. The rate at which I receive new sponsors is not high so this put my progress back almost a year.

Edit: 2 weeks later and almost 1/3rd of the sponsors I had have stopped.

**Comment by squidfunk** [comment]
Some new insights: if you have a webhook installed on your sponsors profile you can find out whether the sponsorship was cancelled by @github-staff by inspecting the `sender` property of the cancellation event:

``` json
{
  "sender": {
    "login": "github-staff",
    "id": 21092374,
    "node_id": "MDQ6VXNlcjIxMDkyMzc0",
    "avatar_url": "https://avatars.githubusercontent.com/u/21092374?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/github-staff",
    "html_url": "https://github.com/github-staff",
    "followers_url": "https://api.github.com/users/github-staff/followers",
    "following_url": "https://api.github.com/users/github-staff/following{/other_user}",
    "gists_url": "https://api.github.com/users/github-staff/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/github-staff/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/github-staff/subscriptions",
    "organizations_url": "https://api.github.com/users/github-staff/orgs",
    "repos_url": "https://api.github.com/users/github-staff/repos",
    "events_url": "https://api.github.com/users/github-staff/events{/privacy}",
    "received_events_url": "https://api.github.com/users/github-staff/received_events",
    "type": "User",
    "site_admin": false
  }
}
```

**Comment by squidfunk** [comment]
I've analyzed the damage – my calculations show that $240k/m in sponsorships were lost in 28 days.
https://twitter.com/squidfunk/status/1643539228574269443

**Reply by bodograumann** [reply]
Thanks for this impact analysis. The lack of publicity and accountability on github's side is astounding.
This also lead to a lot of speculation and accusations, which I am not a fan of.
Still, I'll leave this somewhat related link here, if anybody might have missed it: https://github.blog/2023-04-04-whats-new-with-github-sponsors/

**Comment by Wickedlilang3l** [comment]
I have stripe and then metapay accepts payments for developers as well

**Comment by unitycoder** [comment]
so, stripe is adding paypal payments support,
https://stripe.com/docs/payments/paypal

why couldnt they just tell that in the beginning..
